Sources/ArcGISToolkit/Components/FeatureFormView/Subviews/Inputs/BarcodeScannerInput.swift

```diff
@@ -0,0 +1,196 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+@preconcurrency import AVFoundation
+import SwiftUI
+
+struct BarcodeScannerInput: View {
+    /// The element the input belongs to.
+    private let element: FieldFormElement
+    
+    /// The input configuration of the field.
+    private let input: BarcodeScannerFormInput
+    
+    /// A Boolean value indicating whether the code scanner is presented.
+    @State private var scannerIsPresented = false
+    
+    /// The current barcode value.
+    @State private var value = ""
+    
+    /// Creates a view for a barcode scanner input.
+    /// - Parameters:
+    ///   - element: The input's parent element.
+    init(element: FieldFormElement) {
+        precondition(
+            element.input is BarcodeScannerFormInput,
+            "\(Self.self).\(#function) element's input must be \(BarcodeScannerFormInput.self)."
+        )
+        self.element = element
+        self.input = element.input as! BarcodeScannerFormInput
+    }
+    
+    var body: some View {
+        HStack {
+            Text(value.isEmpty ? String.noValue : value)
+            Spacer()
+            if !value.isEmpty {
+                ClearButton {
+                    value.removeAll()
+                }
+            }
+            Image(systemName: "barcode.viewfinder")
+                .foregroundStyle(.tint)
```

Comment:
While this matches the approved design, it conflicts with the accesory symbol on date elements.

| `.tint` | `.secondary` |
|---|---|
| <img src="https://github.com/user-attachments/assets/638e2886-92b4-4afb-a307-c908004eb7e1"> | <img src="https://github.com/user-attachments/assets/1ec81d46-a4a3-4d32-868d-53dea44a9cb5"> |

Reply:
I think they should match. @dfeinzimer @kaushikrw - any preference as to blue or gray (.tint or .secondary)?

Reply:
I think the gray looks uniform. On kotlin, the tint is either gray/white based on light/dark theme.

Reply:
Secondary/gray is my preference too. The microphone in other system applications is commonly gray.

| Maps | Contacts |
|---|---|
| <img src="https://github.com/user-attachments/assets/10ebca71-e32f-41fb-b871-b9f4c59bd132"> | <img src="https://github.com/user-attachments/assets/51d78d2f-b31e-4e10-949a-505745d98da1"> |

Reply:
Ok, Gray it is. I'll update the design doc.

Reply:
I've updated the design doc images.

Reply:
![](https://devtopia.esri.com/runtime/common-toolkit/blob/main/designs/Forms/BarcodeUI.png)


![](https://devtopia.esri.com/runtime/common-toolkit/blob/main/designs/Forms/BarcodeUI_Filled.png)